core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java

```diff
@@ -215,8 +216,18 @@ private boolean pollForCompletion(BlockingQueue<CompletableFuture<?>> queue, lon
             // This shouldn't block
             future.get(100, TimeUnit.MILLISECONDS);
          } catch (ExecutionException e) {
-            errors.incrementAndGet();
-            log.exceptionPurgingDataContainer(e.getCause());
+            Throwable ce = e.getCause();
+            while (ce instanceof RemoteException) {
+               ce = ce.getCause();
+            }
+            if (ce instanceof org.infinispan.util.concurrent.TimeoutException) {
```

Comment:
shouldn't be inside the "whileW?

Reply:
I was trying to be a bit more specific here as this is the only current case where this happens. Past reviews has made me try to be as specific as possible instead of iterating all causes for example.

Reply:
Actually thinking more about it, I think being outside is most appropriate and more clearly delineates the logic.

Reply:
fair enough :) 

but you have a while-loop: `while (ce instanceof RemoteException)` 
I don't think it is possible to have a RemoteException inside another RemoteException

Reply:
I was just making sure as some other places already do this. I can't say if it is either way, but at least if it is possible this way will work :) and still works if not.